### PR TITLE
adding check for negative zero

### PR DESCRIPTION
**Pull request by abpasaxo**
Adding check for '-0' in case of formatting infinitesimally small negative values. The output should be '0'

### CODE REVIEW

File: src/price-formatting/format.js [hunk 1]
```diff
@@ -199,7 +199,10 @@ function formatPriceParts(numberFormatting, value, decimals, formatFlags, numera
 
     if (isNegative) {
         parts.Post = numberFormatting.negativePost;
-        parts.Pre = numberFormatting.negativePre;
+
+        // Infinitesimally small negative value is rounded to 0, in which case Pre should not be '-'
+        // as '-0' makes no sense, hence the below check
+        parts.Pre = (parts.First === '0' && !parts.Pips) ? '' : numberFormatting.negativePre;
```
Comment (lukeapage) [inline src/price-formatting/format.js]: To be thorough, should we check decipips too?

File: src/price-formatting/format.js [hunk 2]
```diff
@@ -199,7 +199,10 @@ function formatPriceParts(numberFormatting, value, decimals, formatFlags, numera
 
     if (isNegative) {
         parts.Post = numberFormatting.negativePost;
```
Comment (lukeapage) [inline src/price-formatting/format.js]: the same check needs to happen on post - thats for cultures where the - goes after or -1 is represented as (1)

Comment (abpasaxo) [reply src/price-formatting/format.js]: Adding check for post as well.

File: src/price-formatting/format.js (hunk 1 above)
Comment (abpasaxo) [reply src/price-formatting/format.js]: I discussed this with Mayank, decipips won't come untill and unless there is a value for pips. The decipips check would be redundant. Hence skipping it.

Comment (lukeapage) [reply src/price-formatting/format.js]: @abpasaxo I think mayank is wrong .. it’s probably never going to happen but let’s code for it in case

Comment (abpasaxo) [reply src/price-formatting/format.js]: Added check for decipips.

**Comment by lukeapage** [comment]
superceded by https://github.com/SaxoBank/openapi-clientlib-js/pull/63

**Pull request closed without merging**